sdk/resource/auto.go

```diff
@@ -49,31 +51,70 @@ func Detect(ctx context.Context, detectors ...Detector) (*Resource, error) {
 	return r, detect(ctx, r, detectors)
 }
 
+type detectInternalResult struct {
+	detectorIndex int // index in the detector slice
+	resource      *Resource
+	err           error
+}
+
+func detectInternal(ctx context.Context, detectors []Detector) []detectInternalResult {
+	wg := sync.WaitGroup{}
+	resChan := make(chan detectInternalResult)
+	for i := range detectors {
+		wg.Add(1)
+		go func(index int) {
+			defer wg.Done()
+
+			detector := detectors[index]
+			if detector == nil {
+				return
+			}
+
+			r, err := detector.Detect(ctx)
+			resChan <- detectInternalResult{
+				detectorIndex: index,
+				resource:      r,
+				err:           err,
+			}
+		}(i)
+	}
+
+	go func() {
+		wg.Wait()
+		close(resChan)
+	}()
+
+	resultSlice := []detectInternalResult{}
+	for res := range resChan {
+		resultSlice = append(resultSlice, res)
+	}
+
+	slices.SortFunc(resultSlice, func(a, b detectInternalResult) int {
+		return a.detectorIndex - b.detectorIndex
+	})
```

Comment:
because len(resultSlice)!=len(detectors)
if there is nil detector it get filtered i didnt want to change the program behavior from  main branch
and i wanted the resources to be merged in the order of the passed detectors since the merge of resources
 function is not communicative operation

Reply:
It seems like they should be made to be the same length then. This is quite wasteful.

Reply:
It seems like it should be made to be the same length. This is quite wasteful.

Reply:
Since we need to loop another time to do the `Merge` operation anyway, we could check for `nil` in there, which would allow assigning on the index, as suggested.

Reply:
done